--- src/lib/libmosquitto/lib/net_mosq.c.orig	2018-03-01 18:54:29.821064891 +0100
+++ src/lib/libmosquitto/lib/net_mosq.c	2018-03-01 18:55:14.689330569 +0100
@@ -1213,7 +1213,6 @@
 #ifndef WITH_BROKER
 int _mosquitto_socketpair(mosq_sock_t *pairR, mosq_sock_t *pairW)
 {
-#ifdef WIN32
 	int family[2] = {AF_INET, AF_INET6};
 	int i;
 	struct sockaddr_storage ss;
@@ -1317,23 +1316,5 @@
 		return MOSQ_ERR_SUCCESS;
 	}
 	return MOSQ_ERR_UNKNOWN;
-#else
-	int sv[2];
-
-	if(socketpair(AF_UNIX, SOCK_STREAM, 0, sv) == -1){
-		return MOSQ_ERR_ERRNO;
-	}
-	if(_mosquitto_socket_nonblock(sv[0])){
-		COMPAT_CLOSE(sv[1]);
-		return MOSQ_ERR_ERRNO;
-	}
-	if(_mosquitto_socket_nonblock(sv[1])){
-		COMPAT_CLOSE(sv[0]);
-		return MOSQ_ERR_ERRNO;
-	}
-	*pairR = sv[0];
-	*pairW = sv[1];
-	return MOSQ_ERR_SUCCESS;
-#endif
 }
 #endif
